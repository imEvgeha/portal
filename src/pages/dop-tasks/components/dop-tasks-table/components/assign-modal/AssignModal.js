import React, {useEffect} from 'react';
import PropTypes from 'prop-types';
import Select from '@atlaskit/select';
import {connect} from 'react-redux';
import {
    CHANGE_PRIORITY_TITLE,
    TASK_ACTIONS_ASSIGN,
    TASK_ACTIONS_FORWARD,
    PRIORITY_OPTIONS,
} from '../../../../constants';
import {getDopTasksOwners} from '../../../../dopTasksActions';
import {createTaskOwnersSelector} from '../../../../dopTasksSelectors';
import './AssignModal.scss';

const AssignModal = ({getOwners, selectedTasks, tasksOwners, onChange, action}) => {
    useEffect(() => {
        if (action === TASK_ACTIONS_ASSIGN) {
            getOwners(selectedTasks.map(n => n.id));
        } else if (action === TASK_ACTIONS_FORWARD) {
            getOwners([]);
        }
    }, []);

    return (
        <div className="dop-tasks-assign-tasks">
            <label>{`${action} to: `}</label>
            <Select
                placeholder={`Select ${
                    action === CHANGE_PRIORITY_TITLE
                        ? 'priority'
                        : action === TASK_ACTIONS_ASSIGN
                        ? 'Task Owner'
                        : 'Work Queue'
                }`}
                options={action === CHANGE_PRIORITY_TITLE ? PRIORITY_OPTIONS : tasksOwners}
                onChange={val => onChange(val.value)}
            />
        </div>
    );
};

AssignModal.propTypes = {
    getOwners: PropTypes.func.isRequired,
    onChange: PropTypes.func,
    selectedTasks: PropTypes.array,
    tasksOwners: PropTypes.array,
    action: PropTypes.string,
};

AssignModal.defaultProps = {
    onChange: () => null,
    selectedTasks: [],
    tasksOwners: [],
    action: TASK_ACTIONS_ASSIGN,
};

const mapStateToProps = () => {
    const ownersSelector = createTaskOwnersSelector();
    return state => ({
        tasksOwners: ownersSelector(state),
    });
};

const mapDispatchToProps = dispatch => ({
    getOwners: payload => dispatch(getDopTasksOwners(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(AssignModal);
