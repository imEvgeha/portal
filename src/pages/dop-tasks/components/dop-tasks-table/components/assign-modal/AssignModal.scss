.dop-tasks-assign-tasks {
    min-height: 100px;
}
