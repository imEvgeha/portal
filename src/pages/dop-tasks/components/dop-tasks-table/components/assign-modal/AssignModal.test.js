import React from 'react';
import {shallow} from 'enzyme';
import configureStore from 'redux-mock-store';
import AssignModal from './AssignModal';

describe('AssignModal', () => {
    let wrapper = null;

    beforeEach(() => {
        const mockStore = configureStore();
        const store = mockStore({
            dopTasks: {
                tasksOwners: [
                    {
                        firstName: 'test',
                        lastName: 'test',
                        userId: 'test',
                    },
                ],
            },
        });
        wrapper = shallow(<AssignModal store={store} />)
            .shallow()
            .shallow();
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render Select with one option', () => {
        expect(
            wrapper.find('WithAnalyticsContext(WithAnalyticsEvents(AtlaskitSelect))').props().options.length
        ).toEqual(1);
    });
});
