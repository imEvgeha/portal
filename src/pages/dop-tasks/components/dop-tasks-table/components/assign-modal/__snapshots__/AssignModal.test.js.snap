// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignModal should match snapshot 1`] = `
<div
  className="dop-tasks-assign-tasks"
>
  <label>
    Assign to: 
  </label>
  <WithAnalyticsContext(WithAnalyticsEvents(AtlaskitSelect))
    onChange={[Function]}
    options={
      Array [
        Object {
          "firstName": "test",
          "lastName": "test",
          "userId": "test",
        },
      ]
    }
    placeholder="Select Task Owner"
  />
</div>
`;
